query/src/main/scala/filodb/query/exec/PeriodicSamplesMapper.scala

```diff
@@ -87,32 +87,33 @@ final case class PeriodicSamplesMapper(startMs: Long,
           val rdrv = rv.asInstanceOf[RawDataRangeVector]
           val chunkedHRangeFunc = rangeFuncGen().asChunkedH
           val minResolutionMs = rdrv.minResolutionMs
-          if (chunkedHRangeFunc.isInstanceOf[CounterChunkedRangeFunction[_]] && windowLength < minResolutionMs)
+          val extendedWindow = extendLookback(rdrv, windowLength)
+          if (chunkedHRangeFunc.isInstanceOf[CounterChunkedRangeFunction[_]] && extendedWindow < minResolutionMs * 2)
             throw new IllegalArgumentException(s"Minimum resolution of data for this time range is " +
               s"${minResolutionMs}ms. However, a lookback of ${windowLength}ms was chosen. This will not " +
-              s"yield intended results for rate/increase functions since each lookback window contains " +
-              s"lesser than 2 samples. Increase lookback to more than ${minResolutionMs}ms")
-          val windowPlusPubInt = extendLookback(rv, windowLength)
+              s"yield intended results for rate/increase functions since each lookback window can contain " +
+              s"lesser than 2 samples. Increase lookback to more than ${minResolutionMs * 2}ms")
           IteratorBackedRangeVector(rv.key,
             new ChunkedWindowIteratorH(rdrv, startWithOffset, adjustedStep, endWithOffset,
-                    windowPlusPubInt, chunkedHRangeFunc, querySession, histRow), outputRvRange)
+                    extendedWindow, chunkedHRangeFunc, querySession, histRow), outputRvRange)
         }
       case c: ChunkedRangeFunction[_] =>
         source.map { rv =>
           qLogger.trace(s"Creating ChunkedWindowIterator for rv=${rv.key}, adjustedStep=$adjustedStep " +
             s"windowLength=$windowLength")
           val rdrv = rv.asInstanceOf[RawDataRangeVector]
-          val minResolutionMs = rdrv.minResolutionMs
+          val minResolutionMs = rdrv.minResolutionMs * 2
           val chunkedDRangeFunc = rangeFuncGen().asChunkedD
-          if (chunkedDRangeFunc.isInstanceOf[CounterChunkedRangeFunction[_]] && windowLength < rdrv.minResolutionMs)
+          val extendedWindow = extendLookback(rdrv, windowLength)
+          if (chunkedDRangeFunc.isInstanceOf[CounterChunkedRangeFunction[_]] &&
+              extendedWindow < rdrv.minResolutionMs * 2)
             throw new IllegalArgumentException(s"Minimum resolution of data for this time range is " +
               s"${minResolutionMs}ms. However, a lookback of ${windowLength}ms was chosen. This will not " +
-              s"yield intended results for rate/increase functions since each lookback window contains " +
-              s"lesser than 2 samples. Increase lookback to more than ${minResolutionMs}ms")
-          val windowPlusPubInt = extendLookback(rv, windowLength)
+              s"yield intended results for rate/increase functions since each lookback window can contain " +
+              s"lesser than 2 samples. Increase lookback to more than ${minResolutionMs * 2}ms")
```

Comment:
This is 4x of `rdrv.minResolutionMs`, the logic and the message in the two cases are different. I guess what we want is not to change ``minResolutionMs`` but introduce something like ``minExpected:Lookback = minResolutionMs * 2`` and use that in the condition and the exception

Reply:
Fixed